include/seqan3/io/structure_file/format_vienna.hpp

```diff
@@ -215,42 +215,43 @@ class structure_file_input_format_REMOVEME<format_vienna>
         }
 
         // READ STRUCTURE (if present)
+        [[maybe_unused]] size_t structure_length{};
         if constexpr (!detail::decays_to_ignore_v<structure_type>)
         {
             if constexpr (structured_seq_combined)
             {
                 assert(std::addressof(seq) == std::addressof(structure));
                 using alph_type = typename value_type_t<structure_type>::structure_alphabet_type;
-                std::ranges::copy(read_structure<alph_type>(stream_view), begin(structure));
+                std::ranges::copy(read_structure<alph_type>(stream_view, structure_length), begin(structure));
```

Comment:
Please write a comment that we need this length counter for the structured sequence field, because we cannot ensure the length equality through `size(seq)` and `size(structure)`.